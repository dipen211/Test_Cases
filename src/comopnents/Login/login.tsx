import React, { Component } from 'react';

export default class Login extends Component {
    constructor(props: any) {
        super(props);
        this.state = {
            email: '',
            password: ''
        }
    }

    handleChange = (event: any) => {
        this.setState({ [event.target.name]: event.target.value });
    }
    render() {
        return (
            <div className="container-fluid box mt-5">
                <div className="loginbox shadow p-5">
                    <form className="p-2 mt-5">
                        <div className="mb-3">
                            <input id="email" type="text" name="email" onChange={this.handleChange} placeholder="Email/Username" className="text-center" />
                        </div>
                        <div className="mb-3 mt-4">
                            <input id="password" type="password" name="password" onChange={this.handleChange} placeholder="Password" className="text-center" />
                        </div>
                        <div className="Log-button">
                            <button className="btn" style={{ width: '10rem' }}>LOGIN</button>
                        </div>

                    </form>
                </div>
            </div>
        )
    }
}