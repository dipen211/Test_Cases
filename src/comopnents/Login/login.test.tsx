import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Login from './login';
import Adapter from 'enzyme-adapter-react-16';
import { mountToJson } from 'enzyme-to-json';

Enzyme.configure({ adapter: new Adapter() });
describe('Login component tests', () => {
    let wrapper: any;
    beforeEach(() => {
        wrapper = shallow(<Login />);
    });
    it('should match the snapshot', () => {
        expect(mountToJson(wrapper)).toMatchSnapshot();
    });
    it('handle Change', () => {
        wrapper.instance().handleChange({ target: { value: 'test', name: 'email' } });
        wrapper.update();
        expect(wrapper.state('email')).toEqual('test');
    });
    it('should have a btn component', () => {
        expect(wrapper.find('button')).toHaveLength(1);

        expect(wrapper.find('button').text()).toEqual('LOGIN');
    });

    it('should have input for email and password', () => {
        expect(wrapper.find('input#email')).toHaveLength(1);
        expect(wrapper.find('input#password')).toHaveLength(1);
    });

    it('should have an empty email and password state var', () => {
        expect(wrapper.state('email')).toEqual('');
        expect(wrapper.state('password')).toEqual('');
    });
});
