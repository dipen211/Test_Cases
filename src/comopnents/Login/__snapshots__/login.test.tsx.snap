// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component tests should match the snapshot 1`] = `
<div
  className="container-fluid box mt-5"
>
  <div
    className="loginbox shadow p-5"
  >
    <form
      className="p-2 mt-5"
    >
      <div
        className="mb-3"
      >
        <input
          className="text-center"
          id="email"
          name="email"
          onChange={[Function]}
          placeholder="Email/Username"
          type="text"
        />
      </div>
      <div
        className="mb-3 mt-4"
      >
        <input
          className="text-center"
          id="password"
          name="password"
          onChange={[Function]}
          placeholder="Password"
          type="password"
        />
      </div>
      <div
        className="Log-button"
      >
        <button
          className="btn"
          style={
            Object {
              "width": "10rem",
            }
          }
        >
          LOGIN
        </button>
      </div>
    </form>
  </div>
</div>
`;
